server/players.lua

```diff
@@ -64,9 +64,20 @@ RegisterNetEvent('ps-adminmenu:server:SetJob', function(data, selectedData)
     local playerId, Job, Grade = selectedData["Player"].value, selectedData["Job"].value, selectedData["Grade"].value
     local Player = QBCore.Functions.GetPlayer(playerId)
     local name = Player.PlayerData.charinfo.firstname..' '..Player.PlayerData.charinfo.lastname
-
-    QBCore.Functions.Notify(src, locale("jobset", name, Job, Grade), 'success', 5000)
-    Player.Functions.SetJob(tostring(Job), tonumber(Grade))
+    local jobInfo = QBCore.Shared.Jobs[Job]
+    if jobInfo then
+        if jobInfo["grades"][selectedData["Grade"].value] then
+            Player.Functions.SetJob(tostring(Job), tonumber(Grade))
+            if Config.RenewedPhone then
+                exports['qb-phone']:hireUser(tostring(Job), Player.PlayerData.citizenid, tonumber(Grade))
+            end
+            QBCore.Functions.Notify(src, locale("jobset", name, Job, Grade), 'success', 5000)
+        else
+            TriggerClientEvent('QBCore:Notify', source, "Not a valid grade", 'error')
+        end
+    else
+        TriggerClientEvent('QBCore:Notify', source, "Not a valid job", 'error')
+    end
```

Comment:
I think we can utilize guard clauses here to make the code more readable like so:

```suggestion
    local grade = jobInfo["grades"][selectedData["Grade"].value]
    
    if not jobInfo then
        TriggerClientEvent('QBCore:Notify', source, "Not a valid job", 'error')
        return
    end
    
    if not grade then
        TriggerClientEvent('QBCore:Notify', source, "Not a valid grade", 'error')
        return
     end
    
    Player.Functions.SetJob(tostring(Job), tonumber(Grade))
            if Config.RenewedPhone then
                exports['qb-phone']:hireUser(tostring(Job), Player.PlayerData.citizenid, tonumber(Grade))
            end
            
            QBCore.Functions.Notify(src, locale("jobset", name, Job, Grade), 'success', 5000)
```
Also, question about this - it might be correct though. The following:
```lua
exports['qb-phone']:hireUser(tostring(Job), Player.PlayerData.citizenid, tonumber(Grade))
```
is inside a `Config.RenewedPhone` if statement. Is this correct since it triggers a method with qb-phone, but the if statement is about renewed phone?

Reply:
Thats how renewed phone hire people. issue if its not there people that gets hired using the menu will lose their job on relog.

Reply:
> Thats how renewed phone hire people. issue if its not there people that gets hired using the menu will lose their job on relog.

But it should not be `if not Config.RenewedPhone then` (note the not) since it triggers a qb-phone export?

Reply:
Huh? no lol it shouldn't be if not since you need to set the config option to true when using the phone so it uses the export.

Reply:
Ah, I think I get it now. Didn't know how exactly the ecosystem around that was. I'll mark it as completed but needs testing (since I can't do that myself). Thanks for the PR!